### PR TITLE
Adds: the logic for fetching and storing the remote feature flags from BE

### PR DESCRIPTION
**Pull request by AjeshRPai**
Parent #17050 

Dependant FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2518

This PR changes the feature flag logic. The [Firebase remote config](WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt) is swapped with the logic to fetch the config from the back. The feature flags are fetched on every app launch to keep the flags up to date. 

On launching the App, if there are no feature flags in the database then all the flags from the remote config defaults are fetched and stored in the database. Then the remote values are fetched and updated. 

Minor change: This PR updates the order in which the feature flags are shown. It will now be shown based on alphabetical order. 

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt [hunk 1]
```diff
@@ -15,6 +15,8 @@ import org.wordpress.android.util.config.AppConfig.FeatureState
 import javax.inject.Inject
 import javax.inject.Named
 
+const val REMOTE_REFRESH_INTERVAL_IN_HOURS = 12
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt]: @AjeshRPai - a little overkill here, but what if we make TTL a remote value -- could use Firebase for this (defeats the purpose of this project), but would give us the flexibility to adjust. 😜

**Comment by zwarm** [comment]
@AjeshRPai - nice progress. I still need to take it out for a spin, only did a review of the code today

File: WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt]: @zwarm Sorry for the late reply. I was focusing on the Hack week and completely ignored the GitHub notifications. 

> defeats the purpose of this project

Yeah, this was my thought as well when I thought we could make this a Remote value. 😄 . Maybe for some time, we can make this a remote value and later on after trial and error we could define a hardcoded time. How does that sound?

Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt]: @AjeshRPai - We don't have to make it a remote value. Let's not overkill - it's my brain working overtime. :)

File: WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt [hunk 2]
```diff
@@ -37,8 +45,11 @@ class RemoteConfig
     private suspend fun fetchRemoteFlags() {
         Log.e("Refreshing remote flags", " ")
         val response = featureFlagStore.fetchFeatureFlags(
-                deviceId = "12345",
-                identifier = BuildConfig.APPLICATION_ID
+                buildNumber = BuildConfig.VERSION_CODE.toString(),
+                deviceId = preferences.getString(WPCOM_PUSH_DEVICE_UUID, "")?:"",
+                identifier = BuildConfig.APPLICATION_ID,
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt]: @AjeshRPai  - What's the difference between BuidConfig.APPLICATION_ID and packageName from package manager?

Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/util/config/RemoteConfig.kt]: As discussed over sync call the BuidConfig.APPLICATION_ID captures the entire package name with Flavour as well. Am marking the conversation as resolved

**Comment by zwarm** [comment]
👋 @AjeshRPai   - so this Debug View still doesn't match the existing behavior. Let me brain dump.

Excluding manual overrides, there are 3 possible values for every FF.
(1) The build config value
(2) The “FeatureConfig” class  `@Feature` “default value”
(3) The remote value (Firebase or our BE)

I think our issue lies within the confines of the pre-processing for the “default” value. This value appears to be set as the default if a remote field can’t be found in FB. Most of our `FeatureConfig` classes specify a remote field, but do NOT add a remote field into firebase, so I believe this “default value” is used in its place.

**Take this example:**
Class = BloggingPromptsFeatureConfig
Default value = true
BuildConfig value = false
Remote value = does not exist in Firebase

Comparing the value of `blogging_prompts_remote_field `within the debug setting view in the app:
This PR: The field is unchecked
Not this PR: The field is checked

The only difference between the two builds is the value of that default. Confused yet? 😉 

Pull up fun`AppConfig.isEnabled` and read the following comment - it clarifies the use of the build config.

    /**
     * Get the enabled state of a feature flag. If the flag is enabled in the BuildConfig file, it overrides the remote
     * value. The correct approach is to disable a feature flag for a release version and only enable it remotely.
     * Once the feature is ready to be fully released, we can enable the BuildConfig value.
     * @param feature feature which we're checking remotely
     */

This matches the logic we see in `AppConfig.buildFeatureState`. In the above example the buildConfig = false, so when executing `AppConfig.buildFeatureState` we fall into the else statement `remoteConfig.getFeatureState(feature.remoteField, feature.buildConfigValue)` in which this PR returns the false value. My assumption is that we aren't using the "default" value like we did previously.

🤔  I wonder if we should be nulling out remote fields for FF that aren't using a remote companion field? Or they are there "just in case". Either way, if we want to match the current implementations we need to take that default value into account.

**Comment by zwarm** [comment]
@AjeshRPai  - Thanks for making the last round of changes. I was able to match the debug settings in this PR with the debug settings prior to this change. 👍

**Review by zwarm — DISMISSED** [review]
👋 Things appear to be working as expected. There are some log lines that need to be removed, so after you clean it up I'll give things one more go before giving it my 👍 . I already approved the FluxC companion.

**Comment by enejb** [comment]
I updated the Label based on the testing we did yesterday. I think PR is ready to merge :shipit:

**Comment by AjeshRPai** [comment]
@enejb Thanks Enej 🙌🏼 

@zwarm 
I have removed the redundant log messages from the PR. This PR is ready for another re-review. 

I am working on fixing the CI failures. The failure seems to be because of a [wire mock verification](https://console.firebase.google.com/u/1/project/api-project-108380595987/testlab/histories/bh.b3d1e88d8b09c8be/matrices/7563537982526774915/executions/bs.9b85c82f3689d96d/testcases/11/test-cases). Still looking into that.

**Review by enejb — APPROVED** [review]
This works as expected. Thanks for all the changes and making the tests work!

**Pull request merged**